### PR TITLE
A couple of misc bits in ykllvm

### PR DESCRIPTION
**Pull request by vext01**
See commits.

### CODE REVIEW

**Comment by ptersilie** [comment]
Change looks good though I don't really understand this sentence: `Before, the first instruction of a function *defined* the function's arguments.`

File: llvm/lib/Transforms/Yk/LivenessAnalysis.cpp [hunk 1]
```diff
@@ -121,11 +121,11 @@ LivenessAnalysis::LivenessAnalysis(Function *Func) {
 
   // A function implicitly defines its arguments.
   //
-  // To propagate the arguments properly we pretend that the first instruction
-  // in the entry block defines the arguments.
+  // To propagate the arguments properly we say that the function's arguments
```
Comment (ltratt) [inline llvm/lib/Transforms/Yk/LivenessAnalysis.cpp:124]: I can't parse this sentence.

Comment (ptersilie) [reply llvm/lib/Transforms/Yk/LivenessAnalysis.cpp:124]: This is essentially saying, that in order for the function arguments to be live, the liveness algo needs to pretend that the function arguments are incoming values into the first block. It's just the way the algo works.

Comment (ltratt) [reply llvm/lib/Transforms/Yk/LivenessAnalysis.cpp:124]: That's fine: I just can't parse the English ;)

Comment (vext01) [reply llvm/lib/Transforms/Yk/LivenessAnalysis.cpp:124]: Which part Laurie?

Comment (ltratt) [reply llvm/lib/Transforms/Yk/LivenessAnalysis.cpp:124]: The sentence.

Comment (vext01) [reply llvm/lib/Transforms/Yk/LivenessAnalysis.cpp:124]: "incoming to" -> "flow into"?

Comment (ltratt) [reply llvm/lib/Transforms/Yk/LivenessAnalysis.cpp:124]: Edd pointed out that my comments was, ahem, not entirely helpful. Sorry!

One possible rephrasing we came up with is "We need our liveness algorithm to consider the function's arguments live but the algorithm only considers instructions. To ensure that those arguments are considered as live, we make inputs to the function's first instruction."

Comment (vext01) [reply llvm/lib/Transforms/Yk/LivenessAnalysis.cpp]: I had a go using my own words and came up with 395a946d607a06c5738115be354e447a5e5e13a9.

Maybe I spelled it out too hard?

Comment (ltratt) [reply llvm/lib/Transforms/Yk/LivenessAnalysis.cpp]: Works for me!

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Comment by vext01** [comment]
@ltratt ping

**Pull request merged**